go/converters.go

```diff
@@ -39,3 +47,134 @@ func XML(any interface{}) string {
 
 	return string(str)
 }
+
+func SQL(dict DictionaryRepresentable, sqlDialect string) string {
```

Comment:
**HOLY MOLY** there ain't no **way** I'm claiming to remotely understand what's going on here haha. I guess a couple questions I'll pose:

1. Should `sqlDialect` be an "enum" (like `DumpFormat`) instead of just a plain string?
2. Do we need to manually track IDs for everything if we can just generate SQL that uses auto-incrementing columns?
3. Is there any way to better modularize or document this code block to make it less overwhelming lol? The code for writing/serializing dictionaries used to look a lot like this code block until I added methods on each struct (Etymology, Entry, etc.) to convert them from FlatBuffers and back. It broke up the code and made it more modular and easier to understand. I wonder if we could do something similar here. Thoughts?

Another comment about this section is to just make sure you're adding ample whitespace around blocks. A lot of if statements are flush with the statements below them and it hurts readability.

Reply:
To answer these questions in order:
1. Yes I 100% agree we should be using an enum so that we can enforce this function is only called with valid strings w/ type checks.
2. The reason why I kept track of the IDs for everything was for the foreign keys. Although I think omitting the primary key would work I think we'd still need reference to these for the foreign keys.
3. Yeah I apologize for this monstrosity LOL. It got the job done so I kinda chucked it into PR. I will strategize if I can put each dictionary level in a function below the main `SQL` function 💪🏻